cylc/flow/scheduler.py

```diff
@@ -1033,14 +1064,15 @@ def kill_tasks(
         unkillable: List[TaskProxy] = []
         for itask in itasks:
             if itask.state(*TASK_STATUSES_ACTIVE):
-                itask.state_reset(
-                    # directly reset to failed in sim mode, else let
-                    # task_job_mgr handle it
-                    status=(TASK_STATUS_FAILED if jobless else None),
-                    is_held=True,
-                )
-                self.data_store_mgr.delta_task_state(itask)
+                if itask.state_reset(is_held=True):
+                    self.data_store_mgr.delta_task_state(itask)
                 to_kill.append(itask)
+                if jobless:
```

Comment:
Note `jobless` is really `is_sim_mode`:

https://github.com/wxtim/cylc/blob/b31f4ff57f280a40aefec9f593215d046c81ad30/cylc/flow/scheduler.py#L1062

This doesn't matter because you can't kill a skip mode task as it goes straight from waiting to succeeded